### PR TITLE
bugfix: reset bool 'simple' to true when apply to a new IR node

### PR DESCRIPTION
**Pull request by hanw**
Actually change is between line 43 and 58.

I also ran clang-format on this file.

### CODE REVIEW

File: midend/simplifyKey.h [hunk 1]
```diff
@@ -79,13 +84,17 @@ class IsValid : public KeyIsSimple {
...
         if (auto mask = expression->to<IR::BAnd>()) {
             if (mask->right->is<IR::Constant>())
                 expression = mask->left;
             else if (mask->left->is<IR::Constant>())
-                expression = mask->right; }
-        return IsLikeLeftValue::isSimple(expression, ctxt); }
+                expression = mask->right;
+        }
+        auto isSimple = IsLikeLeftValue::isSimple(expression, ctxt);
+        std::cout << "is simple? " << expression << isSimple << std::endl;
```
Comment (mbudiu-vmw) [inline midend/simplifyKey.h]: This should be LOG at most.

File: midend/simplifyKey.h [hunk 2]
```diff
@@ -142,27 +152,35 @@ class TableInsertions {
  * @post all complex table key expressions are replaced with a simpler expression.
  */
 class DoSimplifyKey : public Transform {
-    ReferenceMap*     refMap;
-    TypeMap*          typeMap;
-    KeyIsSimple*      key_policy;
+    ReferenceMap* refMap;
```
Comment (mbudiu-vmw) [inline midend/simplifyKey.h]: I actually liked the previous formatting better.

File: midend/simplifyKey.h [hunk 3]
```diff
@@ -142,27 +152,35 @@ class TableInsertions {
...
     // after the program has been simplified
-    const IR::Node* postorder(IR::MethodCallStatement* statement) override
-    { return doStatement(statement, statement->methodCall); }
-    const IR::Node* postorder(IR::IfStatement* statement) override
-    { return doStatement(statement, statement->condition); }
-    const IR::Node* postorder(IR::SwitchStatement* statement) override
-    { return doStatement(statement, statement->expression); }
-    const IR::Node* postorder(IR::AssignmentStatement* statement) override
-    { return doStatement(statement, statement->right); }
+    const IR::Node* postorder(IR::MethodCallStatement* statement) override {
```
Comment (mbudiu-vmw) [inline midend/simplifyKey.h]: @ChrisDodd won't like this; you are taking 3 times more lines.
I don't think we should run clang-format blindly.

File: midend/simplifyKey.h [hunk 4]
```diff
@@ -53,8 +52,12 @@ class IsLikeLeftValue : public KeyIsSimple, public Inspector {
     void postorder(const IR::Member*) override {}
     void postorder(const IR::PathExpression*) override {}
     void postorder(const IR::ArrayIndex*) override {}
+    profile_t init_apply(const IR::Node* root) override {
```
Comment (mbudiu-vmw) [inline midend/simplifyKey.h]: This change is approved.

**Comment by hanw** [comment]
Given we have more and more people working on the code base, there is a need for a consistent coding style that can be easily enforced. We can try to stay as close to the existing coding style as possible, if we can get that encoded in a clang format file.

**Comment by mbudiu-vmw** [comment]
Given that @ChrisDodd wrote the style guide document for the compiler (see the folder docs) you should work on this with him. Perhaps you want to customize the clang style if it is possible.
The way we stayed consistent was through code reviews.

**Comment by mbudiu-vmw** [comment]
I will approve the changes, but please don't merge until you resolve the style issues.

**Pull request merged**